lib/Commands/BuildSystemCommand.cpp

```diff
@@ -492,7 +492,9 @@ class BasicBuildSystemFrontendDelegate : public BuildSystemFrontendDelegate {
     if (basic::sys::pipe(BasicBuildSystemFrontendDelegate::signalWatchingPipe) < 0) {
       perror("pipe");
     }
-    new std::thread(&BasicBuildSystemFrontendDelegate::signalWaitThread, this);
+    auto* handlerThread = new std::thread(
```

Comment:
Realized on the drive home this 'new' is actually not necessary with the detach and will still be leaking a small object.